tests/server/test_build_logger.py

```diff
@@ -0,0 +1,136 @@
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+import yaml
+
+import pytest
+from deepsparse.server.build_logger import build_logger
+from deepsparse.server.config import ServerConfig
+
+
+yaml_config_1 = """
+num_cores: 2
+num_workers: 2
+loggers:
+    python:
+endpoints:
+    - task: question_answering
+      route: /unpruned/predict
+      model: zoo:nlp/question_answering/bert-base/pytorch/huggingface/squad/base-none
+      batch_size: 1
+      data_logging:
+        - target: pipeline_outputs
+          functions:
+           - func: identity
+             frequency: 5
+           - func: tests/test_data/metric_functions.py:user_defined_identity
+             frequency: 5
+        - target: engine_outputs
+          functions:
+           - func: np.mean
+             frequency: 3"""  # noqa E501
+
+yaml_config_2 = """
+num_cores: 2
+num_workers: 2
+loggers:
+    python: {}
+endpoints:
+    - task: question_answering
+      route: /unpruned/predict
+      model: zoo:nlp/question_answering/bert-base/pytorch/huggingface/squad/base-none
+      batch_size: 1"""
+
+yaml_config_3 = """
+num_cores: 2
+num_workers: 2
+endpoints:
+    - task: question_answering
+      route: /unpruned/predict
+      model: zoo:nlp/question_answering/bert-base/pytorch/huggingface/squad/base-none
+      batch_size: 1"""
+
+yaml_config_4 = """
+num_cores: 2
+num_workers: 2
+loggers:
+    python:
+endpoints:
+    - task: question_answering
+      route: /unpruned/predict
+      model: zoo:nlp/question_answering/bert-base/pytorch/huggingface/squad/base-none
+      batch_size: 1
+      data_logging:
+        - target: pipeline_outputs
+          functions:
+           - func: tests/test_data/metric_functions.py:user_defined_identity
+             frequency: 2
+             target_loggers:
+                - python
+        - target: engine_outputs
+          functions:
+           - func: np.mean
+             frequency: 3"""
+
+yaml_config_5 = """
+num_cores: 2
+num_workers: 2
+loggers:
+    invalid_logger_name:
+endpoints:
+    - task: question_answering
+      route: /unpruned/predict
+      model: zoo:nlp/question_answering/bert-base/pytorch/huggingface/squad/base-none
+      batch_size: 1"""
+
+yaml_config_6 = """
+num_cores: 2
+num_workers: 2
+loggers:
+    python:
+endpoints:
+    - task: question_answering
+      route: /unpruned/predict
+      model: zoo:nlp/question_answering/bert-base/pytorch/huggingface/squad/base-none
+      batch_size: 1
+      data_logging:
+        - target: re:*_outputs
+          functions:
+           - func: tests/test_data/metric_functions.py:user_defined_identity
+             frequency: 2"""
```

Comment:
I interpreted ben's comment from earlier as more akin to:
```suggestion
        - re:*_outputs:
           - func: tests/test_data/metric_functions.py:user_defined_identity
             frequency: 2"""
```

where the key is the target, and the value is the list of functions. So notably, there isn't a target & functions key.

This would change the type of data_logging to
```python
Optional[Dict[str, List[Union[MetricFunctionConfig, str]]]]
```
& would require changes in `build_function_loggers`

@bfineran can you confirm?

Reply:
agreed, yes missed this in review - no need for the outer list